relay-server/src/services/project_cache.rs

```diff
@@ -899,6 +865,42 @@ impl ProjectCacheBroker {
         }
     }
 
+    /// Returns `true` if the project state valid for the [`QueueKey`].
+    ///
+    /// Which includes the own key and the samplig key for the project.
+    fn is_state_valid(&mut self, key: &QueueKey) -> bool {
```

Comment:
Should we add to the docs that this method may trigger a refresh for all projects?